addons/sale_project/models/project.py

```diff
@@ -716,6 +716,11 @@ class ProjectTask(models.Model):
     # Project sharing  fields
     display_sale_order_button = fields.Boolean(string='Display Sales Order', compute='_compute_display_sale_order_button')
 
+    def write(self, vals):
+        if len(self) > 1 and 'sale_line_id' in vals and (len(self.partner_id.ids) != 1 or any(not task.partner_id for task in self)):
```

Comment:
Note that it can be slightly refactored as such, but that's a detail:
```suggestion
        if len(self) > 1 and 'sale_line_id' in vals and not (len(self.partner_id.ids) == 1 and all(task.partner_id for task in self)):
```

Reply:
Ok sure @auon-odoo 
Thank You :blush:

Reply:
Changes done
Thank You :slightly_smiling_face: